javascript/javascript1/week2/index.js

```diff
@@ -0,0 +1,234 @@
+const Male = "male";
+const Female = "female";
+
+function getFullname(firstName, sirName, useFormalName = false, gender = Male) {
+  if (arguments.length < 2) {
+    console.warn("First name and last name must be given");
+    return;
+  }
+  if (typeof firstName != "string" || typeof sirName != "string") {
+    console.warn("First name and last name must be of type string");
+    return;
+  }
+  //remove empty spaces within the name
+  if (firstName.length > 0) {
+    //firstName = firstName.Trim();
+    firstName = firstName.replace(/\s/g, "");
+  }
+  if (sirName.length > 0) {
+    //sirName = sirName.trim();
+    sirName = sirName.replace(/\s/g, "");
+  }
+  if (firstName === "" && sirName === "") {
+    //throw new console.error("First name and sir name should not be empty");
+    console.warn("First name and sir name should not be empty");
+    //throw "First name and sir name should not be empty";
+    return;
+  }
+  if (!(useFormalName === true || useFormalName === false)) {
+    //throw "Invalid parameter, bolean expected";
+    console.warn("Invalid parameter, bolean expected");
+    return;
+  }
+
+  if (useFormalName) {
+    if (gender === Male) {
+      return `Lord ${firstName} ${sirName}`;
+    } else if (gender === Female) {
+      return `Lady ${firstName} ${sirName}`;
+    }
+  }
+  return `${firstName} ${sirName}`;
+}
+
+const fullname1 = getFullname("Reenu", "Abraham");
+if (fullname1 != undefined) {
+  console.log(fullname1);
+}
+const fullname2 = getFullname("Benjamin", "Hughes", 5);
+console.log(fullname2);
+const fullname3 = getFullname("Sofia", "John", true, "abcd");
+console.log(fullname3);
+const fullname4 = getFullname("Catrine", "", true, Female);
+if (fullname4 != undefined) {
+  console.log(fullname4);
+}
+
+const fullname5 = getFullname(" ", "", true, Female);
+if (fullname4 != undefined) {
+  console.log(fullname4);
+}
+
+//Ex 2
+const weekDay = {
+  0: "Sunday",
+  1: "Monday",
+  2: "Tuesday",
+  3: "Wednesday",
+  4: "Thursday",
+  5: "Friday",
+  6: "Saturday",
+};
+
+function getEventWeekday(noOfdaysFromToday) {
+  const day = new Date();
+  day.setDate(day.getDate() + noOfdaysFromToday);
+  return weekDay[day.getDay()];
+}
+
+console.log(`Event day is ${getEventWeekday(40)}`);
+
+//Ex3
+
+function getClothingRecommendation(temperatureInCelcius) {
+  if (temperatureInCelcius < 0) {
+    return "4 layers of winter clothes";
+  }
+  if (temperatureInCelcius <= 10) {
+    return "3 layers of winter clothes";
+  }
+  if (temperatureInCelcius > 10 && temperatureInCelcius <= 15) {
```

Comment:
do you need to check if it is bigger than 10?
```suggestion
  if (temperatureInCelcius <= 15) {
```